src/snowflake/snowpark/_internal/open_telemetry.py

```diff
@@ -55,7 +55,11 @@ def open_telemetry_context_manager(func, dataframe):
             except Exception as e:
                 logger.warning(f"Error when acquiring span attributes. {e}")
             finally:
-                yield
+                try:
+                    yield
+                except Exception as e:
+                    cur_span.set_status(trace.Status(trace.StatusCode.ERROR, str(e)))
```

Comment:
How verbose is the `str(e)`? I was thinking just a simple "Python Exception" should suffice, or if there is short description for python exception class type. The actual exception stack trace will be recorded in logs. 
https://opentelemetry-python.readthedocs.io/en/latest/api/trace.status.html#opentelemetry.trace.status.Status